src/model.js

```diff
@@ -99,10 +99,15 @@ const BookshelfModel = ModelBase.extend({
    *   be the singular form of this model's {@link Model#tableName tableName},
    *   followed by `_id` / `_{{{@link Model#idAttribute idAttribute}}}`.
    *
+   * @param {string=} foreignKeyTarget
+   *
+   *   Column in the `Target` model's table which `foreignKey` references, if other
+   *   than `Target` model's `id` / `{@link Model#idAttribute idAttribute}`.
+   *
    * @returns {Model}
    */
-  hasOne(Target, foreignKey) {
-    return this._relation('hasOne', Target, {foreignKey}).init(this);
+  hasOne(Target, foreignKey, foreignKeyTarget) {
+    return this._relation('hasOne', Target, {foreignKey, foreignKeyTarget}).init(this);
```

Comment:
In my [comment](https://github.com/tgriesser/bookshelf/pull/1397#issuecomment-250721474) I meant changing this to

```
{foreignKey, targetIdAttribute: foreignKeyTarget}
```